python/pylibvw.cc

```diff
@@ -261,6 +105,35 @@ vw_ptr my_initialize_with_log(std::string args, py_log_wrapper_ptr py_log)
   return boost::shared_ptr<vw>(foo);
 }
 
+vw_ptr my_initialize_with_pyred(std::string args, py_log_wrapper_ptr py_log, py::object with_reduction)
+{
+  if (args.find_first_of("--no_stdin") == std::string::npos) args += " --no_stdin";
+
+  trace_message_t trace_listener = nullptr;
+  void* trace_context = nullptr;
+
+  if (py_log)
+  {
+    trace_listener = (py_log_wrapper::trace_listener_py);
+    trace_context = py_log.get();
+  }
+
+  if (with_reduction)
+  {
+    auto ext_binding = std::unique_ptr<wrapper::external_binding>(new py_cpp_bridge(&with_reduction));
+    auto learner_builder = VW::make_unique<custom_builder_with_binding>(std::move(ext_binding));
+
+    return boost::shared_ptr<vw>(
+        VW::initialize_with_builder(args, nullptr, false, trace_listener, trace_context, std::move(learner_builder)));
+  }
+  else
+  {
+    return boost::shared_ptr<vw>(VW::initialize(args, nullptr, false, trace_listener, trace_context));
+  }
+
+  return boost::shared_ptr<vw>();
```

Comment:
This return will never be reached. Perhaps just remove `else` and let the return on line 131 replace this